nubis/files/duo.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# We run this to configure sshd_config with puppet and bounce the service
+# its split out here because we were running into puppet memory allocation errors
+# when it tries to bounce sshd so this gets to live here now
+
+OS=$(lsb_release --id | awk '{print $3}')
+
+case "${OS}" in
+    Ubuntu)
+        SSH_SERVICE="ssh"
+    ;;
+    CentOS)
+        SSH_SERVICE="sshd"
+    ;;
+    *)
+        # If not supported we just assume ssh service is just ssh
+        SSH_SERVICE="ssh"
+    ;;
+esac
+
+puppet apply /etc/duo/duo_sshd_config_runtime.pp && systemctl restart "${SSH_SERVICE}"
```

Comment:
I thought the discussed plan was to avoid using puppet at all in favor of
```
augtool set /files/etc/ssh/sshd_config/AuthenticationMethods publickey,keyboard-interactive 
```

Reply:
My recollection is the opposite we decided or at least we agreed that just splitting the files this way is fine. I would rather we do it this way in order to avoid having to install an additional tool like augtool on the host